lib/hobbes/events/events.C

```diff
@@ -91,6 +91,21 @@ void registerInterruptHandler(const std::function<void()>& fn) {
   (*epClosures)[-1] = c;
 }
 
+namespace {
+int handleEvents(int timeoutMS) {
+  struct epoll_event evts[64];
+  const int fds = epoll_wait(threadEPollFD(), evts, sizeof(evts)/sizeof(evts[0]), timeoutMS);
+  if (fds > 0) {
+    for (int fd = 0; fd < fds; ++fd) {
+      auto* c = reinterpret_cast<eventcbclosure*>(evts[fd].data.ptr);
+      (c->fn)(c->fd);
+      resetMemoryPool();
+    }
+  }
+  return fds;
+}
+}
+
 bool stepEventLoop(int timeoutMS) {
   while (true) {
```

Comment:
```c++
while(not stopFn()) {
  ...
}
```

Reply:
`epoll_wait` blocks forever on default `timeoutMS` value `-1`, `stopFn` won't get the second chance to be evaluated

Reply:
The timeout delay used by `epoll_wait` can be changed by pushing a timer event; there's an api for that